### PR TITLE
Upgrade guide for rec-4.3

### PR DESCRIPTION
**Pull request by omoerbeek**
### Checklist

I have:
- [X] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [ ] compiled this code
- [ ] tested this code
- [X] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/recursordist/docs/upgrade.rst [hunk 1]
```diff
@@ -7,8 +7,42 @@ When upgrading several versions, please read **all** notes applying to the upgra
...
 ------------------------
 
+Lua Netmask class methods changed
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 - Netmask class methods ``isIpv4`` and ``isIpv6`` have been deprecated in Lua, use :func:`Netmask.isIPv4` and :func:`Netmask.isIPv6` instead. In C++ API these methods have been removed.
 
+``socket-dir`` changed
+^^^^^^^^^^^^^^^^^^^^^^
+The default :ref:`setting-socket-dir` has changed to include ``pdns-recursor`` in the path.
+For non-chrooted setups, it is now whatever is passed to ``--with-socketdir`` during configure (``/var/run`` by default) plus ``pdns_recursor``.
```
Comment (mnordhoff) [inline pdns/recursordist/docs/upgrade.rst]:
```suggestion
For non-chrooted setups, it is now whatever is passed to ``--with-socketdir`` during configure (``/var/run`` by default) plus ``pdns-recursor``.
```
Either I missed some complexity, or the directory always has a hyphen, not an underscore, right? (I think the _file_ has an underscore.)

File: pdns/recursordist/docs/upgrade.rst [hunk 2]
```diff
@@ -7,8 +7,42 @@ When upgrading several versions, please read **all** notes applying to the upgra
...
+The systemd unit-file is updated to reflect this change and systemd will automatically create the directory with the proper permissions.
+The packaged sysV init-script also creates this directory.
+For other operating systems, update your init-scripts accordingly.
+
+Systemd service and permissions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The systemd service-file that is installed no longer uses the ``root`` user to start.
+It uses the user and group set with the ``--with-service-user`` and ``--with-service-group`` switches during
+configuration, "pdns" by default.
+This could mean that PowerDNS Recursor cannot read its configuration or zone-file data.
```
Comment (mnordhoff) [inline pdns/recursordist/docs/upgrade.rst]: Should "or zone-file data" be changed or removed? Recursor does support RPZ, but still.

File: pdns/recursordist/docs/upgrade.rst (hunk 1 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/upgrade.rst]: How I hate these gratuitous differences...

File: pdns/recursordist/docs/upgrade.rst (hunk 2 above)
Comment (omoerbeek) [reply pdns/recursordist/docs/upgrade.rst]: I'll update to mention recursor relevant files.

File: pdns/recursordist/docs/upgrade.rst [hunk 3]
```diff
@@ -7,8 +7,42 @@ When upgrading several versions, please read **all** notes applying to the upgra
...
+For non-chrooted setups, it is now whatever is passed to ``--with-socketdir`` during configure (``/var/run`` by default) plus ``pdns-recursor``.
+The systemd unit-file is updated to reflect this change and systemd will automatically create the directory with the proper permissions.
+The packaged sysV init-script also creates this directory.
+For other operating systems, update your init-scripts accordingly.
+
+Systemd service and permissions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+The systemd service-file that is installed no longer uses the ``root`` user to start.
+It uses the user and group set with the ``--with-service-user`` and ``--with-service-group`` switches during
+configuration, "pdns" by default.
```
Comment (pieterlexis) [inline pdns/recursordist/docs/upgrade.rst]: The user is pdns on debian, pdns-recursor on CentOS

**Pull request merged**